### PR TITLE
Mock managers

### PR DESCRIPTION
**Pull request by mpangburn**
Features:
- Pump simulator with controllable reservoir and battery, ability to force errors on bolus/temp/suspend/resume.
- CGM simulator with controllable glucose production model (constant or sine curve) and composable effects (glucose noise, random outliers, random errors).
- Simulator configuration via settings; ability to clear out simulator data from HealthKit.

Notes:
- Strings are deliberately not localized; this is a development tool and not for production.
- Bolus and suspend states don't make use of transitional "bolusing" or "suspending" states.

### CODE REVIEW

File: LoopKitUI/View Controllers/GlucoseEntryTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    private lazy var glucoseFormatter: NumberFormatter = {
+        let quantityFormatter = QuantityFormatter()
+        quantityFormatter.setPreferredNumberFormatter(for: glucoseUnit)
+        return quantityFormatter.numberFormatter
+    }()
+
+    public var glucose: HKQuantity? {
+        get {
+            guard let doubleValue = value.flatMap(Double.init) else {
```
Comment (ps2) [inline LoopKitUI/View Controllers/GlucoseEntryTableViewController.swift]: Is `Double(value)` more readable than `value.flatMap(Double.init)`?

File: MockKit/MockCGMManager.swift [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    public var rawState: RawStateValue {
+        return [
+            "mockSensorState": mockSensorState.rawValue,
+            "dataSource": dataSource.rawValue
+        ]
+    }
+
+    public var appURL: URL? {
```
Comment (ps2) [inline MockKit/MockCGMManager.swift]:
```
    public let appURL: URL? = nil

    public let providesBLEHeartbeat = false

    public let managedDataInterval: TimeInterval? = nil

    public let shouldSyncToRemoteService = false
```

**Review by ps2 — APPROVED** [review]
Just nits. Looks great!

File: LoopKitUI/View Controllers/GlucoseEntryTableViewController.swift (hunk 1 above)
Comment (mpangburn) [reply LoopKitUI/View Controllers/GlucoseEntryTableViewController.swift]: Arguably, yeah. The gain of `flatMap` is that it saves you an unwrap. The alternative here is:
```swift
guard let value = value, let doubleValue = Double(value) else { /* ... */ }
```

**Pull request merged**